packages/headless/doc-parser/src/entity.ts

```diff
@@ -1,29 +1,58 @@
-export interface Entity {
+type EntityKind = 'primitive' | 'object' | 'function';
+
+interface Kind<T extends EntityKind> {
+  kind: T;
+}
+
+interface Name {
   name: string;
+}
+
+interface Description {
   desc: string;
+}
+
+interface Type {
   type: string;
+}
+
+interface Optional {
   isOptional: boolean;
 }
 
-export interface ObjEntity extends Entity {
+export interface Entity
+  extends Kind<'primitive'>,
+    Name,
+    Description,
+    Type,
+    Optional {}
+
+export interface ObjEntity
+  extends Kind<'object'>,
+    Name,
+    Description,
+    Type,
+    Optional {
   members: AnyEntity[];
   isTypeExtracted: boolean;
   typeName: string;
 }
 
-export interface FuncEntity {
-  name: string;
-  desc: string;
+export interface FuncEntity extends Kind<'function'>, Name, Description {
   params: AnyEntity[];
   returnType: string | ObjEntity | FuncEntity;
 }
 
 export type AnyEntity = Entity | ObjEntity | FuncEntity;
 
-export function isObjectEntity(entity: unknown): entity is ObjEntity {
-  if (typeof entity !== 'object') {
-    return false;
-  }
+interface UnknownEntity {
+  kind: EntityKind;
+}
+
+export function isObjectEntity(entity: UnknownEntity): entity is ObjEntity {
+  return entity.kind === 'object';
+}
 
-  return !!entity && 'members' in entity;
```

Comment:
I felt checking if a certain key was present was fragile. If it was renamed, this check would fail and typescript would not alert us. To have stricter typing, I added a `kind` property to all entities,